### PR TITLE
[inference]fix import bug and delete down useless init

### PR DESCRIPTION
**Pull request by CjhHa1**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
This PR fix the import bugs and delete useless inits

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Review by Xu-Kai — APPROVED** [review]

**Review by yuanheng-zhao — APPROVED** [review]
LGTM, LLAMA benchmark works as usual

File: colossalai/inference/tensor_parallel/modeling/_utils.py [hunk 1]
```diff
@@ -1,10 +1,51 @@
...
+    # NTK  ref: https://www.reddit.com/r/LocalLLaMA/comments/14lz7j5/ntkaware_scaled_rope_allows_llama_models_to_have/
+    try:
+        ntk_alpha = float(os.environ.get("INFER_NTK_ALPHA", 1))
+        assert ntk_alpha >= 1
+        if ntk_alpha > 1:
+            print(f"Note: NTK enabled, alpha set to {ntk_alpha}")
+        max_seq_len *= ntk_alpha
+        base = base * (ntk_alpha ** (self.head_dim_ / (self.head_dim_ - 2)))  # Base change formula
+    except:
+        pass
```
Comment (FrankLeeeee) [inline colossalai/inference/tensor_parallel/modeling/_utils.py:40]: This is weird. If just `pass` when an error is raised, then why even use `try` statement?

File: colossalai/inference/tensor_parallel/modeling/_utils.py [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
     return
+
+
+def init_to_get_rotary(self, base=10000, use_elem=False):
+    self.config.head_dim_ = self.config.hidden_size // self.config.num_attention_heads
+    if not hasattr(self.config, "rope_scaling"):
+        rope_scaling_factor = 1.0
+    else:
+        rope_scaling_factor = self.config.rope_scaling.factor if self.config.rope_scaling is not None else 1.0
+    if hasattr(self.config, "max_sequence_length"):
```
Comment (FrankLeeeee) [inline colossalai/inference/tensor_parallel/modeling/_utils.py:23]: There should be an empty between two `if` statement blocks?

File: colossalai/inference/tensor_parallel/modeling/_utils.py [hunk 3]
```diff
@@ -1,10 +1,51 @@
...
+    if use_elem:
+        n_elem //= 2
+
+    inv_freq = 1.0 / (base ** (torch.arange(0, n_elem, 2, device="cpu", dtype=torch.float32) / n_elem))
+    t = torch.arange(max_seq_len + 1024 * 64, device="cpu", dtype=torch.float32) / rope_scaling_factor
+    freqs = torch.outer(t, inv_freq)
+
+    self._cos_cached = torch.cos(freqs).to(torch.float16).cuda()
+    self._sin_cached = torch.sin(freqs).to(torch.float16).cuda()
+    return
```
Comment (FrankLeeeee) [inline colossalai/inference/tensor_parallel/modeling/_utils.py:51]: you don't need to write `return` if you do not  return anything.

File: colossalai/inference/tensor_parallel/modeling/llama.py [hunk 4]
```diff
@@ -5,12 +5,9 @@
 from transformers.models.llama.modeling_llama import LlamaAttention, LlamaDecoderLayer, LlamaModel, LlamaRMSNorm
 
 from colossalai.inference.tensor_parallel.batch_infer_state import BatchInferState
-from colossalai.kernel.triton import (
-    copy_kv_cache_to_dest,
-    llama_context_attn_fwd,
-    rotary_embedding_fwd,
-    token_attention_fwd,
-)
+from colossalai.kernel.triton import llama_context_attn_fwd, rotary_embedding_fwd, token_attention_fwd
```
Comment (FrankLeeeee) [inline colossalai/inference/tensor_parallel/modeling/llama.py:8]: try and except? what if the user does not have triton installed?

File: colossalai/inference/tensor_parallel/modeling/_utils.py [hunk 5]
```diff
@@ -1,10 +1,51 @@
...
 from colossalai.kernel.triton.copy_kv_cache_dest import copy_kv_cache_to_dest
 
 
 def copy_kv_to_mem_cache(layer_id, key_buffer, value_buffer, context_mem_index, mem_manager):
     copy_kv_cache_to_dest(key_buffer, context_mem_index, mem_manager.key_buffer[layer_id])
     copy_kv_cache_to_dest(value_buffer, context_mem_index, mem_manager.value_buffer[layer_id])
     return
+
+
+def init_to_get_rotary(self, base=10000, use_elem=False):
```
Comment (FrankLeeeee) [inline colossalai/inference/tensor_parallel/modeling/_utils.py:17]: docstring is required to explain this function, e.g. what does it do, what models can it apply to, etc.

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 2 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:23]: done

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 1 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:40]: `NTK` is a environment variable and if we do not use it, nothing changes.

Comment (FrankLeeeee) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:40]: The correct way to write this is:

```python
ntk_alpha = os.environ.get("INFER_NTK_ALPHA", None)

if ntk_alpha is not None:
      ntk_alpha = float(ntk_alpha)
      assert ...
      if ...


```

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 2 above)
Comment (FrankLeeeee) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:23]: It seems the change is not applied.

File: colossalai/inference/tensor_parallel/modeling/llama.py (hunk 4 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/llama.py:8]: The check for `triton` is done in `_init_`

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 5 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:17]: for sure!

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 3 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:70]: sure

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 2 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py]: Now changes are applied, try see it

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 1 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py]: done!

File: colossalai/inference/tensor_parallel/modeling/_utils.py (hunk 5 above)
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/_utils.py:25]: done

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**